### PR TITLE
[#5159] Rails 6 upgrade

### LINKED ISSUE #5159: Upgrade to Rails 6.0
Requires https://github.com/mysociety/alaveteli/issues/5069.
Requires Ruby 2.5 https://blog.bigbinary.com/2019/04/09/rails-6-requires-ruby-2-5-or-newer.html
Requires Postgres 9.4+ https://blog.bigbinary.com/2019/04/10/rails-6-drops-support-for-postgresql-less-than-9-3.html

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Fixes #5159 

## What does this do?

1. Upgrade Alaveteli to Rails 6.0
2. Refactor gettext/locale loading to resolve app booting issue in development environment. 

## Notes to reviewer

This PR doesn't prep for the Rails 6.1 release as the CI will be red and with how Github actions works we can't allow failures without it putting red crosses everywhere and emailing us. We don't really want that so lets prep in a separate PR and fix CI there.

### CODE REVIEW

File: lib/alaveteli_localization.rb [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+      repos = paths.map do |path|
+        FastGettext::TranslationRepository.build('app', path: path, type: :po)
+      end
+      AlaveteliLocalization.set_default_text_domain('app', repos)
+
+      I18n::Backend::Simple.send(:include, I18n::Backend::Fallbacks)
+
+      AlaveteliLocalization.set_default_locale_urls(
+        AlaveteliConfiguration.include_default_locale_in_urls
+      )
```
Comment (gbp) [inline lib/alaveteli_localization.rb:24]: There was some minor refactoring here but should be equivalent to what was in `config/initializers/fast_gettext.rb`

**Review by garethrees — COMMENTED** [review]
> 0ca6063

Nice!

What are we doing about the new [autoloading](https://guides.rubyonrails.org/v6.0/upgrading_ruby_on_rails.html#autoloading) stuff? Is that getting handled in https://github.com/mysociety/alaveteli/issues/5851 or do they need to be addressed here?

Not much change so fine in principle. I've booted the app successfully but haven't tested further than that.

File: lib/alaveteli_localization.rb [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
-require 'alaveteli_localization/underscorred_locale'
-
 class AlaveteliLocalization
+  class Railtie < Rails::Railtie
+    config.before_configuration do
+      require 'alaveteli_localization/locale'
+      require 'alaveteli_localization/hyphenated_locale'
+      require 'alaveteli_localization/underscorred_locale'
+
+      require 'routing_filters'
```
[the remark this answers is not in the document]
Comment (gbp) [reply lib/alaveteli_localization.rb:8]: https://github.com/mysociety/alaveteli/blob/36c1c748bd2466a9366288cedffacf8438ace1e5/lib/alaveteli_localization.rb#L100

This is called within `set_locales`.

File: lib/alaveteli_localization.rb (hunk 1 above)
Comment (gbp) [reply lib/alaveteli_localization.rb]: Is!

**Comment by gbp** [comment]
> What are we doing about the new [autoloading](https://guides.rubyonrails.org/v6.0/upgrading_ruby_on_rails.html#autoloading) stuff? Is that getting handled in #5851 or do they need to be addressed here?

This isn't using Zeitwork yet. We would need load defaults for Rails "6.0" in `config/application.rb` to enable

**Pull request merged**